airflow/executors/executor_types.py

```diff
@@ -0,0 +1,147 @@
+
+TaskInstanceKey = Tuple[str, str, datetime, int]
+
+# State of the task instance.
+# Stores string version of the task state.
+TaskInstanceState = Tuple[TaskInstanceKey, str]
+
+# Command to execute - might be either string or list of strings
+# with the same semantics as subprocess.Popen
+CommandType = Union[str, List[str]]
+
+
+# Task that is queued. It contains all the information that is
+# needed to run the task.
+#
+# Tuple of: command, priority, Queue, SimpleTaskInstance
+class SimpleTaskInstance:
+    """
+    Simplified Task Instance.
+
+    Used to send data between processes via Queues.
+
+    """
+    def __init__(self, ti: TaskInstance):
+        self._dag_id: str = ti.dag_id
+        self._task_id: str = ti.task_id
+        self._execution_date: datetime = ti.execution_date
+        self._start_date: datetime = ti.start_date
+        self._end_date: datetime = ti.end_date
+        self._try_number: int = ti.try_number
+        self._state: str = ti.state
+        self._executor_config: Any = ti.executor_config
+        self._run_as_user: Optional[str] = None
+        if hasattr(ti, 'run_as_user'):
+            self._run_as_user = ti.run_as_user
+        self._pool: Optional[str] = None
+        if hasattr(ti, 'pool'):
+            self._pool = ti.pool
+        self._priority_weight: Optional[int] = None
+        if hasattr(ti, 'priority_weight'):
+            self._priority_weight = ti.priority_weight
+        self._queue: Column = ti.queue
+        self._key = ti.key
+
+    # pylint: disable=missing-docstring
+    @property
+    def dag_id(self) -> str:
+        return self._dag_id
+
+    @property
+    def task_id(self) -> str:
+        return self._task_id
+
+    @property
+    def execution_date(self) -> datetime:
+        return self._execution_date
+
+    @property
+    def start_date(self) -> datetime:
+        return self._start_date
+
+    @property
+    def end_date(self) -> datetime:
+        return self._end_date
+
+    @property
+    def try_number(self) -> int:
+        return self._try_number
+
+    @property
+    def state(self) -> str:
+        return self._state
+
+    @property
+    def pool(self) -> Any:
+        return self._pool
+
+    @property
+    def priority_weight(self) -> Optional[int]:
+        return self._priority_weight
+
+    @property
+    def queue(self) -> Column:
+        return self._queue
+
+    @property
+    def key(self) -> TaskInstanceKey:
+        return self._key
+
+    @property
+    def executor_config(self):
+        return self._executor_config
+
+    @provide_session
+    def construct_task_instance(self, session=None, lock_for_update=False):
+        """
+        Construct a TaskInstance from the database based on the primary key
+
+        :param session: DB session.
+        :param lock_for_update: if True, indicates that the database should
+            lock the TaskInstance (issuing a FOR UPDATE clause) until the
+            session is committed.
+        """
```

Comment:
`:returns: the task instance constructed`
and return type is missing.

Reply:
Yep.